### PR TITLE
add itemWorldMapUpdateRenderRadiusWorld(No/Has)Ceiling a configuable value

### PR DESCRIPTION
**Pull request by jeffry1829**
make itemWorldMapUpdateRenderRadiusWorldNoCeiling/HasCeiling a configuable value

### CODE REVIEW

**Comment by MiniDigger** [comment]
Whats the usecase?

**Comment by EbonJaeger** [comment]
That's a _really_ long name, and I have absolutely no idea what it's supposed to do just by looking at it.

**Comment by jeffry1829** [comment]
> Whats the usecase?

Add a option to reduce lag caused by net.minecraft.server.v1_15_R1.ItemWorldMap.a()
image

**Comment by jeffry1829** [comment]
> That's a _really_ long name, and I have absolutely no idea what it's supposed to do just by looking at it.

Maybe change to ItemMapUpdateRadius(No/Has)Ceiling?

**Comment by jeffry1829** [comment]
Ah, I'll explain what this PR does.
Item Map currently have update radius 128 for Overworld, and 64 for worlds has ceiling
We would need to call getChunkAtWorldCoords for (128*2)^2, i.e. 65535 times per updating.
Adding a option to limit the radius may be useful

**Comment by Black-Hole** [comment]
To me it seems that there is chunk loading involved. So not loading chunks there should fix it much better.

**Comment by jeffry1829** [comment]
> To me it seems that there is chunk loading involved. So not loading chunks there should fix it much better.

You mean loading the chunk but don't ticking it? But the update is very frequent

File: Spigot-Server-Patches/0440-make-itemWorldMapUpdateRenderRadiusWorldNoCeiling-Ha.patch [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
++            int i1 = MathHelper.floor(entity.locZ() - (double) k) / i + 64; // Paper - positive z axis of item map
++            // Paper start - make itemWorldMapUpdateRenderRadiusWorldNoCeiling/HasCeiling a configuable value
++            int j1 = world.paperConfig.itemWorldMapUpdateRenderRadiusWorldHasCeiling / i;
++            
++            if (world.worldProvider.g()) { // Paper - world.worldProvider.hasCeiling() - OBFHELPER
++            	j1 = world.paperConfig.itemWorldMapUpdateRenderRadiusWorldNoCeiling / i;
+             }
+-
++            // Paper end
+             WorldMap.WorldMapHumanTracker worldmap_worldmaphumantracker = worldmap.a((EntityHuman) entity);
```
Comment (electronicboy) [inline Spigot-Server-Patches/0440-make-itemWorldMapUpdateRenderRadiusWorldNoCeiling-Ha.patch:50]: everything below here seems unneeded?

File: Spigot-Server-Patches/0440-make-itemWorldMapUpdateRenderRadiusWorldNoCeiling-Ha.patch [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+-            int j1 = 128 / i;
+-
+-            if (world.worldProvider.g()) {
+-                j1 /= 2;
++            int l = MathHelper.floor(entity.locX() - (double) j) / i + 64; // Paper - positive x axis of item map
++            int i1 = MathHelper.floor(entity.locZ() - (double) k) / i + 64; // Paper - positive z axis of item map
++            // Paper start - make itemWorldMapUpdateRenderRadiusWorldNoCeiling/HasCeiling a configuable value
++            int j1 = world.paperConfig.itemWorldMapUpdateRenderRadiusWorldHasCeiling / i;
++            
++            if (world.worldProvider.g()) { // Paper - world.worldProvider.hasCeiling() - OBFHELPER
```
Comment (electronicboy) [inline Spigot-Server-Patches/0440-make-itemWorldMapUpdateRenderRadiusWorldNoCeiling-Ha.patch:45]: OBFHELPERS should be added to the class being called as per the contrib docs, not as comments in the method, but, this seems unneeded, nothing is changing here?

Looks like this has actually been mapped now, however

File: Spigot-Server-Patches/0440-make-itemWorldMapUpdateRenderRadiusWorldNoCeiling-Ha.patch [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
++            	j1 = world.paperConfig.itemWorldMapUpdateRenderRadiusWorldNoCeiling / i;
```
Comment (electronicboy) [inline Spigot-Server-Patches/0440-make-itemWorldMapUpdateRenderRadiusWorldNoCeiling-Ha.patch:46]: `if (hasCieling) { //noCieling value` ? This seems reversed

**Review by electronicboy — CHANGES_REQUESTED** [review]

**Comment by MiniDigger** [comment]
@jeffry1829 do you plan on addressing the reviews?

**Comment by Owen1212055** [comment]
Closing PR, as it seems that this has gone inactive.

If you're still interested in continuing this PR, feel free to rebase and leave a comment and we can reopen it for you. 😄

**Pull request closed without merging**